### PR TITLE
Call _check_arraylike for jnp.linalg & jnp.fft functions

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #7737

This is a breaking change, but we've long been moving toward this tighter requirement for inputs across the `jax.numpy` APIs. See the linked issue for discussion of this.

### CODE REVIEW

**Review by sharadmv — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/linalg.py [hunk 1]
```diff
@@ -42,13 +42,15 @@ def _H(x: ArrayLike) -> Array:
...
 def cholesky(a: ArrayLike) -> Array:
+  _check_arraylike("jnp.linalg.cholesky", a)
   a, = _promote_dtypes_inexact(jnp.asarray(a))
   return lax_linalg.cholesky(a)
 
 @_wraps(np.linalg.svd)
 @partial(jit, static_argnames=('full_matrices', 'compute_uv', 'hermitian'))
 def svd(a: ArrayLike, full_matrices: bool = True, compute_uv: bool = True,
         hermitian: bool = False) -> Union[Array, Tuple[Array, Array, Array]]:
+  _check_arraylike("jnp.linalg.cholesky", a)
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py:53]: Done

File: jax/_src/third_party/numpy/linalg.py [hunk 2]
```diff
@@ -107,101 +110,102 @@ def tensorsolve(a, b, axes=None):
...
-        result = _multi_dot(arrays, order, 0, n - 1, precision)
-
-    # return proper shape
-    if ndim_first == 1 and ndim_last == 1:
-        return result[0, 0]  # scalar
-    elif ndim_first == 1 or ndim_last == 1:
-        return result.ravel()  # 1-D
-    else:
-        return result
+  _check_arraylike('jnp.linalg.cond', *arrays)
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/third_party/numpy/linalg.py:113]: Done

**Review by sharadmv — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,9 @@ PLEASE REMEMBER TO CHANGE THE '..main' WITH AN ACTUAL TAG in GITHUB LINK.
   back to CPU. Set `JAX_PLATFORMS=''` to override this behavior and automatically
   choose an available backend (the original default), or set `JAX_PLATFORMS=cpu`
   to always use CPU regardless of if the TPU is available.
+  * Functions in {mod}`jax.numpy.linalg` and {mod}`jax.numpy.fft` now uniformly
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply CHANGELOG.md:19]: Done

File: jax/_src/numpy/fft.py [hunk 4]
```diff
@@ -42,6 +42,7 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
               s: Optional[Shape], axes: Optional[Sequence[int]],
               norm: Optional[str]) -> Array:
   full_name = "jax.numpy.fft." + func_name
+  _check_arraylike(full_name, a)
 
   # TODO(jakevdp): call check_arraylike
```
Comment (soraros) [inline jax/_src/numpy/fft.py:47]: This is no longer needed, right?

Comment (jakevdp) [reply jax/_src/numpy/fft.py]: Done, thanks

**Pull request merged**